app/validators/value_source_validator.py

```diff
@@ -68,7 +68,7 @@ def _validate_answer_source_reference(self, identifiers, selector=None):
             if identifier not in answers_with_context:
                 self.add_error(self.ANSWER_REFERENCE_INVALID, identifier=identifier)
 
-            if selector:
+            if selector and identifier in answers_with_context:
```

Comment:
Does the fact that the identifier doesn't exist get picked up later on?. If not, we should raise an error here, as we are otherwise masking the issue.

Reply:
`Identifier` doesn't exist gets picked up at line 68.   I don't understand what issue is getting masked? if the `identifier` doesn't exist , you can't get the `answer_type `for `selector` hence subsequent checks for `answer_type` are irrelevant.

Reply:
Right, missed line 68. So since that checks if it exists. This condition can be simplified and just be `elif selector:`